iocage_lib/ioc_start.py

```diff
@@ -1167,3 +1268,183 @@ def find_bridge_mtu(self, bridge):
         ).split()
 
         return membermtu[5]
+
+    def _command(self, command, *args, fib=None, jail=None):
+        cmd = [command] + list(args)
+        if jail is not None:
+            cmd = ['jexec', jail] + cmd
+        if fib is not None:
+            cmd = ['setfib', fib] + cmd
+        iocage_lib.ioc_common.logit({
+            "level": "DEBUG",
+            "message": f"running command: {cmd}"
+        })
+        try:
+            output = iocage_lib.ioc_common.checkoutput(
+                cmd,
+                stderr=su.STDOUT
+            )
+        except su.CalledProcessError as err:
+            err.output = err.output.decode('utf-8')
+            raise
+
+        return output
+
+    # Wrapper functions enhance readability of long
+    # lists of command and make it easier to add future
+    # conditionals based on command args.
+    def __route__(self, *args, fib=None, jail=None):
+        return self._command('route', *args, fib=fib, jail=jail)
+
+    def __ifconfig__(self, *args, fib=None, jail=None):
+        return self._command('ifconfig', *args, fib=fib, jail=jail)
+
+    def __ngctl__(self, command, *args):
+        return self._command('ngctl', command, *args)
+
+    def get_ng_bridge_nextlink(self, bridge, start=0):
+        nextlink = start
+        while True:
+            try:
+                self.__ngctl__('msg', f"{bridge}:", 'getstats', f"{nextlink}")
+            except su.CalledProcessError:
+                return nextlink
+
+            nextlink += 1
+
+    def get_ng_bridge_members(self, bridge):
+        members = self.__ngctl__('show', f"{bridge}:").splitlines()
+
+        member_exp = re.compile(
+            r'\s*'
+            r'(?P<hook>link(?P<hookindex>\d+))\s+'
+            r'(?P<node>\S+)\s+'
+            r'(?P<nodetype>\S+)\s+'
+            r'(?P<nodeid>\d+)\s+'
+            r'(?P<nodehook>\S+)\s*$'
+        )
+        members = [member.groupdict()
+                    for member in map(member_exp.fullmatch, members)
+                    if member is not None]
+
+        return sorted(members, key=lambda member: int(member['hookindex']))
+
+    def exists_ng_bridge_member(self, bridge, iface):
+        return iface in [member['node'] for member
+                         in self.get_ng_bridge_members(bridge)]
+
+    def find_ng_bridge_mtu(self, bridge):
+        if self.exists_ng_node(bridge):
+            memberif = self.get_ng_bridge_members(bridge)
+        else:
+            # If the bridge doesn't exist, it is
+            # the same as a bridge with 0 members.
+            memberif = []
+
+        if not memberif:
+            return '1500'
+
+        membermtu = self.__ifconfig__(memberif[0]['node']).split()
+
+        return membermtu[5]
+
+    def get_ng_nodetype(self, iface):
+        try:
+            nodetype = self.__ngctl__("show", f"{iface}:").split()
+
+            if nodetype[0] == "Name:" and nodetype[2] == "Type:":
+                return nodetype[3]
```

Comment:
Seeing an else here, is it possible that those indexes don't exist at all ? I am not very familiar with `ngctl`. So if you think there's a possibility that might happen, we better be safe then sorry ;)